### ISSUE TITLE
Missing imports for many to many relations

### ISSUE THREAD

**Issue by mevinDhun** [mean-expert-official/loopback-sdk-builder#87]
I have a _Company_ model which has a many to many relation with _EboardUser_ through _CompanyUser_.

Company.json
`
"relations": {
    "users": {
      "type": "hasMany",
      "model": "EboardUser",
      "foreignKey": "idCompany",
      "through": "CompanyUser"
    }
}
`

When i generated the sdk, i got the following method in the _Company_ model:
```javascript
public linkUsers(id: any, fk: any, data: CompanyUser = undefined) {
    let method: string = "PUT";
    let url: string = LoopBackConfig.getPath() + "/" + LoopBackConfig.getApiVersion() +
    "/Companies/:id/users/rel/:fk";
    let routeParams: any = {
      id: id,
      fk: fk
    };
    let postBody: any = {
      data: data
    };
    let urlParams: any = {};
    let result = this.request(method, url, routeParams, urlParams, postBody);
    return result;
  }
```
But the import is missing _CompanyUser_  (error TS2304: Cannot find name 'CompanyUser'.)
```javascript
import {
  Company,
  EboardUser,
  MeetingType,
  CompanyDoc,
  LoopBackFilter
} from '../../models/index';
```

Same error occurs in _EboardUser_ file ( error TS2304: Cannot find name 'CompanyUser'):
```javascript
public linkCompanies(id: any, fk: any, data: CompanyUser = undefined) {
    let method: string = "PUT";
    let url: string = LoopBackConfig.getPath() + "/" + LoopBackConfig.getApiVersion() +
    "/EboardUsers/:id/companies/rel/:fk";
    let routeParams: any = {
      id: id,
      fk: fk
    };
    let postBody: any = {
      data: data
    };
    let urlParams: any = {};
    let result = this.request(method, url, routeParams, urlParams, postBody);
    return result;
  }
```

**Comment by jonathan-casarrubias** [comment]
@mevinDhun thanks for reporting, I have 2 questions

1.- Is the CompanyUser private model?

2.- If it is not private, it is at least even generated within sdk/models and sdk/services

**Comment by mevinDhun** [comment]
@jonathan-casarrubias  no it is not private. It has been generated and is available along with the other models.

**Comment by jonathan-casarrubias** [comment]
@mevinDhun which version of the SDK builder are you using? I can not replicate this issue, but I refactored the method that build the services method properties and there is a validation that if a model could not be imported, then it should not be assigned as method param type, in which you should see something like:

````js
public linkCompanies(id: any, fk: any, data: any = undefined) {
}
````
This is a validation when no target class was found, I fixed that in version 8 or 8.1 so its a new fix, please check your version and see if the issue persists.

Cheers
jon

**Comment by mevinDhun** [comment]
I am using version 2.0.0-rc.8.1. I have create a sample app to replicate this issue:
[https://github.com/mevinDhun/loopbacksdk-many-test](https://github.com/mevinDhun/loopbacksdk-many-test)

**Comment by jonathan-casarrubias** [comment]
gr8, let me verify the issue

**Comment by jonathan-casarrubias** [comment]
@mevinDhun actually you may not know but you found 2 different issues deep in the core of the builder.

One of the issues was really specific to models that contains the word "any" like Comp{any}, etc. So it was not really because of the type of relation but the name of the models.

But there was actually another issue only related to hasManyThrough relations, since loopback does not list the through as related model, it was not being imported. So I made some tweaks to also add through models to the list.

This issue is fixed and will be published under version RC9

Cheers!
Jon

**Comment by dooddy** [comment]
I had the same issue. So waiting for rc9. Thank you for your work

**Comment by sprypradeep** [comment]
There is another issue snicked in due to this. I have a model school, and two sguides (school has many) and guides (school has many thru sguides), now with 9.1 I get Sguide model imported but the model definition file is skipped (as its private). guide and school are not private, so those models get generated.

Another condition in the fold.
```

    "guides": {
      "type": "hasMany",
      "model": "guide",
      "foreignKey": "schoolId",
      "through": "sguide"
    },
    "sguides": {
      "type": "hasMany",
      "model": "sguide",
      "foreignKey": ""
    },

```
```
import {
...
  Sguide,
...
}
.....
  public updateByIdSguides(id: any, fk: any, data: Sguide = undefined) {


```

**Comment by jonathan-casarrubias** [comment]
@sprypradeep thanks for reporting, I will verify... now that some models are coming from a different place I forgot to verify if these were also private, I will verify and send a fix ASAP.

Cheers
Jon

**Comment by jonathan-casarrubias** [comment]
A new condition has been added so Through models that are private won't be loaded. This will be released under version 2.0.0-stable
